script/update-tasks.sh

```diff
@@ -0,0 +1,217 @@
+
+
+  cat <<-EOF > "$task_path"
+# auto generated by script/update-tasks.sh
+# DO NOT EDIT: use the script instead
+# source: $task_url
+#
+---
+$(curl -sLf "$task_url" |
+  sed -e 's|^kind: Task|kind: ClusterTask|g' \
+      -e "s|^\(\s\+\)workingdir:\(.*\)|\1workingDir:\2|g"  )
+EOF
+
+ # NOTE: helps when the original and the generated need to compared
+ # curl -sLf "$task_url"  -o "$task_path.orig"
+
+}
+
+change_task_image() {
+  local dest_dir="$1"; shift
+  local version="${1//./-}"; shift
+
+  local task="$1"; shift
+  local task_path="$dest_dir/${task}/${task}-task.yaml"
+  local task_path_version="$dest_dir/${task}/${task}-$version-task.yaml"
+
+  local expr=$1; shift
+  local image=$1; shift
+
+  sed \
+      -i "s'$expr.*'$image'" \
+      $task_path
+
+  sed \
+      -i "s'$expr.*'$image'" \
+      $task_path_version
+}
+
+get_tasks() {
+  local dest_dir="$1"; shift
+  local version="${1//./-}"; shift
+
+  local catalog="$1"; shift
+  local catalog_version="$1"; shift
+
+  local -n tasks=$1
+
+
+  info "Downloading tasks from catalog $catalog to $dest_dir directory"
+  for t in ${!tasks[@]} ; do
+    # task filenames do not follow a naming convention,
+    # some are taskname.yaml while others are taskname-task.yaml
+    # so, try both before failing
+    local task_url="$catalog/$catalog_version/task/$t/${tasks[$t]}/${t}.yaml"
+    echo "$catalog/$catalog_version/task/$t/${tasks[$t]}/${t}.yaml"
+    mkdir -p "$dest_dir/$t/"
+    local task_path="$dest_dir/$t/$t-task.yaml"
+
+    download_task  "$task_path" "$task_url"  ||
+      die 1 "Failed to download $t"
+
+    create_version "$task_path" "$t" "$version"  ||
+      die 1  "failed to convert $t to $t-$version"
+  done
+}
+
+
+create_version() {
+  local task_path="$1"; shift
+  local task="$1"; shift
+  local version="$1"; shift
+  local task_version_path="$(dirname $task_path)/$task-$version-task.yaml"
+
+  sed \
+    -e "s|^\(\s\+name:\)\s\+\($task\)|\1 \2-$version|g"  \
+    $task_path  > "$task_version_path"
+}
+
+
+
+main() {
+
+
+  local catalog_version=${1:-''}
+  [[ -z "$catalog_version"  ]] && usage "missing catalog_version"
+  shift
+
+  local dest_dir=${1:-''}
+  [[ -z "$dest_dir"  ]] && usage "missing destination directory"
+  shift
+
+  local version=${1:-''}
+  [[ -z "$version"  ]] && usage "missing task_version"
+  shift
+
+  mkdir -p "$dest_dir" || die 1 "failed to create ${dest_dir}"
+
+  dest_dir="$dest_dir/addons/02-clustertasks"
+  mkdir -p "$dest_dir" || die 1 "failed to create catalog dir ${catalog_dir}"
+
+  get_tasks "$dest_dir" "$version"  \
+    "$TEKTON_CATALOG"   "$catalog_version"  TEKTON_CATALOG_TASKS
+
+  get_tasks "$dest_dir" "$version"  \
+    "$OPENSHIFT_CATALOG"   "$catalog_version"  OPENSHIFT_CATALOG_TASKS
+
+  change_task_image "$dest_dir" "$version"  \
+    "buildah"  "quay.io/buildah"  \
+    "registry.redhat.io/rhel8/buildah@sha256:6a68ece207bc5fd8db2dd5cc2d0b53136236fb5178eb5b71eebe5d07a3c33d13"
```

Comment:
can we add a comment here that how did we get digest sha?

Reply:
updated the script